src/nomad_simulations/schema_packages/properties/energies.py

```diff
@@ -58,29 +90,87 @@ def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
         super().normalize(archive, logger)
 
 
-class ChemicalPotential(PhysicalProperty):
+#! The only issue with this structure is that total energy will never be a sum of its contributions,
+#! since kinetic energy lives separately, but I think maybe this is ok?
+class TotalEnergy(BaseEnergy):
     """
-    Free energy cost of adding or extracting a particle from a thermodynamic system.
+    Physical property section describing the total energy of a (sub)system.
     """
 
-    # ! implement `iri` and `rank` as part of `m_def = Section()`
-
-    iri = 'http://fairmat-nfdi.eu/taxonomy/ChemicalPotential'
-
-    value = Quantity(
-        type=np.float64,
-        unit='joule',
-        description="""
-        The value of the chemical potential.
-        """,
-    )
+    contributions = SubSection(sub_section=EnergyContribution.m_def, repeats=True)
 
     def __init__(
         self, m_def: 'Section' = None, m_context: 'Context' = None, **kwargs
     ) -> None:
         super().__init__(m_def, m_context, **kwargs)
-        self.rank = []
         self.name = self.m_def.name
 
     def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
         super().normalize(archive, logger)
+
+
+# ? Separate quantities for nuclear and electronic KEs?
+class KineticEnergy(BaseEnergy):
+    """
+    Physical property section describing the kinetic energy of a (sub)system.
```

Comment:
no need to repeat "Physical property section". You can try to do some definition better. Same for PotentialEnergy and TotalEnergy